cpp/include/raft/selection/col_wise_sort.cuh

```diff
@@ -0,0 +1,346 @@
+                    CUDA_CHECK(layoutSortOffset(dSegmentOffsets, n_columns, numSegments, stream));
+
+                    CUDA_CHECK(cub::DeviceSegmentedRadixSort::SortPairs(workspacePtr,
+                                                                        workspaceSize,
+                                                                        in,
+                                                                        sortedKeys,
+                                                                        dValuesIn,
+                                                                        out,
+                                                                        totalElements,
+                                                                        numSegments,
+                                                                        dSegmentOffsets,
+                                                                        dSegmentOffsets + 1,
+                                                                        0,
+                                                                        sizeof(InType) * 8,
+                                                                        stream));
+                }
+            } else {
+                // batched per row device wide sort
+                if (workspaceSize == 0 || !workspacePtr) {
+                    OutType* tmpValIn = nullptr;
+
+                    // first call is to get size of workspace
+                    CUDA_CHECK(cub::DeviceRadixSort::SortPairs(
+                            workspacePtr, workspaceSize, in, sortedKeys, tmpValIn, out, n_columns));
+                    bAllocWorkspace = true;
+
+                    if (!sortedKeys)
+                        workspaceSize += raft::alignTo(sizeof(InType) * (size_t)n_columns, memAlignWidth);
+
+                    workspaceSize += raft::alignTo(sizeof(OutType) * (size_t)n_columns, memAlignWidth);
+                } else {
+                    size_t workspaceOffset   = 0;
+                    bool userKeyOutputBuffer = true;
+
+                    if (!sortedKeys) {
+                        userKeyOutputBuffer = false;
+                        sortedKeys          = reinterpret_cast<InType*>(workspacePtr);
+                        workspaceOffset     = raft::alignTo(sizeof(InType) * (size_t)n_columns, memAlignWidth);
+                        workspacePtr        = (void*)((size_t)workspacePtr + workspaceOffset);
+                    }
+
+                    OutType* dValuesIn = reinterpret_cast<OutType*>(workspacePtr);
+                    workspaceOffset    = raft::alignTo(sizeof(OutType) * (size_t)n_columns, memAlignWidth);
+                    workspacePtr       = (void*)((size_t)workspacePtr + workspaceOffset);
+
+                    // layout idx
+                    CUDA_CHECK(layoutIdx(dValuesIn, 1, n_columns, stream));
+
+                    for (int i = 0; i < n_rows; i++) {
```

Comment:
Any reason for `i` to be an `int` as opposed to a `std::size_t`? It looks like `i` is an index, and keeping it as a `std::size_t` can help avoid unexpected signed/unsigned comparison issues.

Also, we should probably default to using `++i` unless we're using the output.